scheduler/match.go

```diff
@@ -47,10 +52,54 @@ func MemoryAvailable(v float64) ogle.Matcher {
 	return &resourceMatcher{"mem", v}
 }
 
+func (m *attributeMatcher) Matches(o interface{}) (err error) {
+	offer := o.(*mesos.Offer)
+	matched := int(0)
+
+	for _, constraint := range m.SlaveConstraints {
```

Comment:
If you combine it all into the AllOf matcher, doesn't that just move the loop of the `task.SlaveConstraints` to the `createMatcher` function (L97)? If that's not what you mean, feel free to take the PR and run with it.